image/descriptor.go

```diff
@@ -67,8 +79,8 @@ func (d *descriptor) validate(w walker) error {
 			return nil
 		}
 
-		digest, err := filepath.Rel("blobs", filepath.Clean(path))
-		if err != nil || d.normalizeDigest() != digest {
+		digest, err := filepath.Rel(filepath.Join("blobs", d.algo()), filepath.Clean(path))
+		if err != nil || d.hash() != digest {
```

Comment:
“Digest” is [algo + hash][1].  Can we call the `digest` variable `filename` or something to avoid confusing folks by comparing it to `d.hash()`?

[1]: https://github.com/opencontainers/image-spec/blob/259eb6b0f533254f25884c2cbee73f1e8e6e290b/descriptor.md#digests-and-verification